api/src/main/java/org/pmiops/workbench/api/RuntimeController.java

```diff
@@ -173,10 +175,51 @@ private DbWorkspace lookupWorkspace(String workspaceNamespace) throws NotFoundEx
         workspaceNamespace, firecloudWorkspaceName, WorkspaceAccessLevel.WRITER);
     workspaceService.validateActiveBilling(workspaceNamespace, firecloudWorkspaceName);
 
-    return ResponseEntity.ok(
-        leonardoMapper.toApiRuntime(
-            leonardoNotebooksClient.getRuntime(
-                workspaceNamespace, userProvider.get().getRuntimeName())));
+    try {
+      return ResponseEntity.ok(
+          leonardoMapper.toApiRuntime(
+              leonardoNotebooksClient.getRuntime(
+                  workspaceNamespace, userProvider.get().getRuntimeName())));
+    } catch (NotFoundException e) {
+      if (!workbenchConfigProvider.get().featureFlags.enableCustomRuntimes) {
+        throw e;
+      }
+
+      List<LeonardoListRuntimeResponse> runtimes =
+          leonardoNotebooksClient.listRuntimesByProject(workspaceNamespace);
+      if (runtimes.isEmpty()) {
+        throw e;
+      }
+
+      LeonardoListRuntimeResponse mostRecentRuntime =
+          runtimes.stream()
+              .sorted(
+                  (a, b) ->
+                      b.getAuditInfo()
```

Comment:
It's possible there are some degenerate entries missing this  - I would make this sort method immune to null auditInfo and createdDate (treat these as older)